### PR TITLE
fix: update e2e promo tests to use multi-stage text checks

### PR DESCRIPTION
**Pull request by jamesgorrie**
Adds checks at the multiple stages of buying a product with a promo.

- Asserts the promo text on the tier card
- Asserts the total text on the checkout
- Asserts the "Buy" button text
- Asserts the promo text on the thank you page

This has picked up a bug that I was using the full amount for the promo text on the thank you page. Nice ✨ .

Lastly we add the annual promo example and header.

The script names for the e2e tests now follow `test-{env}` as it was a tiny bit fiddly to remember what it was.

[Trello card](https://trello.com/c/6EA1c4B7/705-3-tier-promotion-playwright-test-improve)

## Screenshots
Screenshot 2024-03-26 at 14 41 03

### CODE REVIEW

File: support-e2e/tests/tieredCheckout.test.ts [hunk 1]
```diff
@@ -34,12 +34,20 @@ const testDetailsPromo = [
 		frequency: 'Monthly',
 		promoCode: 'PLAYWRIGHT_TEST_SPLUS_MONTHLY',
 		expectedPromoText: '£8/month for the first 3 months, then £10/month',
+		expectedCheckoutTotalText: '£10 £8/month',
```
Comment (jamesgorrie) [inline support-e2e/tests/tieredCheckout.test.ts:37]: I haven't really found a way to work out how to check for <s>strikethrough<s>.

Comment (jamesgorrie) [reply support-e2e/tests/tieredCheckout.test.ts:37]: I might see if we can add some useful content and check for that similar to Option 3 in [this article](https://www.webaxe.org/strikethrough-html-accessibility/).

**Review by johnduffell — APPROVED** [review]
nice work on the tidyup and extra fixes 👏

**Review by paul-daniel-dempsey — COMMENTED** [review]
Looks Good

**Pull request merged**